csv-generator/csv_generator/consumers/__init__.py

```diff
@@ -42,20 +53,9 @@ def feed_consumers(file_list: List[str], output_dir: str, create_date: str,
 
     consumers = _init_consumers(create_date, output_dir, zip_file_name)
 
-    for data_file in file_list:
-        LOGGER.debug('consuming: ', data_file)
-        root = etree.parse(os.path.join(zip_dir, data_file))
+    for source_file in file_list:
+        LOGGER.debug('consuming: ', source_file)
+        root = etree.parse(os.path.join(zip_dir, source_file))
 
         for consumer in consumers:
-            consumer.process(root, xml_file_name=data_file)
-
-
-def _init_consumers(create_date: str, output_dir: str, zip_file_name: str) -> List[BaseXMLConsumer]:
-    """Initialize consumer classes with required values.
-
-    :param create_date: str
-    :param output_dir: str
-    :param zip_file_name: str
-    :return: list
-    """
-    return [CONSUMERS[consumer](create_date, zip_file_name, output_dir) for consumer in CONSUMERS]
+            consumer.process(root, source_file_name=source_file)
```

Comment:
This should join the zip filename with the xml filename if it is part of a zip, e.g. `test.zip/file.xml` (this is so that we know where it came from, the same XML filename may appear in multiple zip files when it is updated)

Reply:
see https://github.com/elifesciences/data-pipeline/pull/114/files#diff-1b0994fdc2d5d06388e4291a7f05248eR47

the zipname is part of the class state, when the row is written the final source path is determined then

Reply:
I am slightly confused. The tests don't seem to reflect that (I added a comment one example). And I just ran it locally from `csv-generator`:

```bash
pipenv run python -m csv_generator.process_xml_zip --source-zip ../example-data/xml-set-1.zip --output-dir ../output
```
The CSVs seem to contain the XML filename only, no reference to `xml-set-1.zip/` in them as far as I can see?

Reply:
Okay, that seems to be fixed now. Thank you.